src/components/AnimatedLink.tsx

```diff
@@ -15,13 +15,13 @@ export default function AnimatedLink({
   const Link = asATag ? "a" : NextLink;
   return (
     <Link
-      className="group flex items-baseline text-xl text-red transition duration-300 hover:text-grey-900"
+      className="group flex items-center text-xl text-red transition hover:text-grey-900"
       href={href}
       target={isExternalLink(href) ? "_blank" : undefined}
     >
       {children}
-      <div className="relative left-0 ml-4 inline-flex h-8 w-8 items-center justify-center rounded-lg border border-red transition-all duration-300 group-hover:ml-3 group-hover:border-grey-900 group-hover:bg-grey-900">
-        <UpRightArrow className="[&_path]:transition-[stroke] [&_path]:duration-300 [&_path]:group-hover:stroke-white" />
+      <div className="relative left-0 ml-4 inline-flex h-8 w-8 items-center justify-center rounded-lg border border-red transition-all group-hover:ml-3 group-hover:border-grey-900 group-hover:bg-grey-900">
+        <Icon name="arrow" className="transition text-red group-hover:text-white -rotate-45 w-6 h-6" />
```

Comment:
by default icons are the same color as the text they appear in, since this is what we want 80% of the time. when we want to change the color we just need to set a text color style on the icon.